dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/GitHubSourceCodeRepo.java

```diff
@@ -213,8 +214,15 @@ public void setLicenseInformation(Entry entry, String gitRepository) {
         // retrieval of directory content is cached as opposed to retrieving individual files
         String fullPathNoEndSeparator = FilenameUtils.getFullPathNoEndSeparator(fileName);
         // but tags on quay.io that do not match github are costly, avoid by checking cached references
-        GHRef[] refs = repo.getRefs();
-        if (Lists.newArrayList(refs).stream().noneMatch(ref -> ref.getRef().contains(reference))) {
+
+        GHRef[] branchesAndTags;
+        try {
+            branchesAndTags = getBranchesAndTags(repo);
+        } catch (IOException ex) {
```

Comment:
See my other comment, but I think it would be less disruptive not to have a catch here.

Reply:
This is slightly different than my previous comment, but after looking at it more closely, I think you should let this throw, no catch.

`repo.getRefs` can throw an IOException; `getBranchesAndTags` can also throw an IOException -- just let the handling stay the same. Long term, your approach may be right, but I think for the hotfix we should minimize control-flow changes.